src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateClosedLaborAccountsCsvServiceImpl.java

```diff
@@ -0,0 +1,253 @@
+        try {
+            sqlDate = getDateTimeService().convertToSqlDate(computedFromDate.toString());
+        } catch (ParseException pe) {
+            LOG.error("getJavaSqlDateForComputedDate: ParseException caught for computedFromDate = " + computedFromDate.toString());
+            sqlDate = null;
+        }
+        return sqlDate;
+    }
+    
+    private LocalDate computeDateBeingDaysPreviousFromToday(int daysPrevious) {
+        LOG.info("computeDateBeingDaysPreviousFromToday: daysPrevious = " + daysPrevious);
+        DateTime todayAsDateTime = new DateTime(getDateTimeService().getCurrentDate());
+        DateTime datePreviousAsDateTime = todayAsDateTime.minusDays(daysPrevious);
+        LOG.info("computeDateBeingDaysPreviousFromToday: datePreviousAsDateTime = " + datePreviousAsDateTime.toLocalDate().toString());
+        return datePreviousAsDateTime.toLocalDate();
+    }
+    
+    private void writeClosedLaborAccountsToCsvFormattedFile(Map<String, Date> fileDataRangeSpecified) {
+        LOG.info("writeClosedLaborAccountsToCsvFormattedFile: Prior to ensuring directory to hold outbound file exists.");
+        ensureOutbundDirectoryExists();
+        
+        LOG.info("writeClosedLaborAccountsToCsvFormattedFile: Prior to obtaining data with SQL query for date range. This may take a while.");
+        List<LaborClosedAccount> laborClosedAccountsData = getClosedLaborAccountsByDateRangeDao().obtainLaborClosedAccountsDataFor(fileDataRangeSpecified);
+        
+        LOG.info("writeClosedLaborAccountsToCsvFormattedFile: Have the data. Will now write it out.");
+        writeClosedAccountsToCsvFile(laborClosedAccountsData);
+    }
+    
+    private void ensureOutbundDirectoryExists() {
+        try {
+            FileUtils.forceMkdir(new File(getCsvLaborClosedAccountsExportDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoryExists: Could not create file destination directory. Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void writeClosedAccountsToCsvFile(List<LaborClosedAccount> laborClosedAccountsDataList) {
+        String fullyQualifiedOutputFileName = generateFullyQualifiedOutputFileName();
+        LOG.info("writeClosedAccountsToCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedOutputFileName);
+        File outputFile = new File(fullyQualifiedOutputFileName);
+        FileWriter outputFileWriter;
+        try {
+            outputFileWriter = new FileWriter(outputFile);
+            for (LaborClosedAccount closedAccount : laborClosedAccountsDataList) {
+                outputFileWriter.write(closedAccount.toCsvString());
+                outputFileWriter.write(KFSConstants.NEWLINE);
+            }
+            outputFileWriter.close();
+        } catch (IOException io) {
+            LOG.error("writeClosedAccountsToCsvFile: Caught IOException = " + io.getMessage());
```

Comment:
instead of logging just the message, I would suggest logging the full exception object.  

Also, if the file can't be written, should the job error?  As written, I think the exception is just logged, but the problem isn't percolated back up.

Reply:
Changing to log io object and bubble up IOException to fail batch job when exception occurs.